lib/restaurant.rb

```diff
@@ -0,0 +1,66 @@
+require_relative 'menu.rb'
+require_relative 'send_sms.rb'
+
+class Restaurant
+  attr_reader :menu
+  attr_reader :basket
+
+  def initialize(menu, sender)
+    @menu = menu
+    @sender = sender
+    @basket = {}
+  end
+
+  def show_menu
+    menu.list
+  end
+
+  def order(dish, portions = 1)
+    fail "Sorry, #{dish} aren't on the menu" unless menu.list.include?(dish)
```

Comment:
I like the fail that you have added here.

Reply:
Thanks - it covers an edgecace so that the customer can't get too carried away making an invalid order and break the 'total' / 'basket_summary' functions.